agent/connect/ca/provider_consul.go

```diff
@@ -438,8 +439,10 @@ func (c *ConsulProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 		return "", fmt.Errorf("error encoding certificate: %s", err)
 	}
 
+	cert := strings.TrimSuffix(buf.String(), "\n")
```

Comment:
Maybe a comment to say we don't expect a missing newline, but we are doing this because other providers had the problem and we want to be cautious.